src/Iact1dUnbinnedLkl.cc

```diff
@@ -889,6 +889,78 @@ Int_t Iact1dUnbinnedLkl::ReaddNdESignal(TString filename)
   return status;
 }
 
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+// 
+// Read multiple dN_i/dE for signal from file in
+// the Segue Stereo input format produced by Jelena
+// and their corresponding branching ratios BR_i.
+// Then perform a linear combination
+// dN/dE = sum(BR_i * dN_i/dE).
+// Replacement of existing file is allowed
+// Return 0 in case of success
+//        1 if file is not found
+//
+Int_t Iact1dUnbinnedLkl::ReaddNdESignal(Int_t nFiles, TString* filenames, Float_t* branchingRatios)
+{
+  // Initialize the array of pointers and other variables
+  TFile* dNdESignalFile[nFiles];
+  TH1F*  hProvdNdESignal[nFiles];
+  TH1F*  hProvdNdESignalTemp;
+  TH1F*  linCHdNdESignal;
+
+  Int_t status      = 0;
+  Int_t nBins       = 0;
+  Double_t lowEdge  = 0.0;
+  Double_t highEdge = 0.0;
+  // open file and look for histograms
+  for(Int_t iFile=0;iFile<nFiles;iFile++)
+    {
+      dNdESignalFile[iFile]  = new TFile(filenames[iFile]);
+      hProvdNdESignalTemp = (TH1F*) dNdESignalFile[iFile]->Get("hdNdE");
+      // check that dN/dE histograms are found
+      if(!hProvdNdESignalTemp) status=1;
```

Comment:
Like this, if just the last file contains a valid histogram, the status is set to 1.
status should be 0 (and should be returned by the function) if ANY of the files contain no valid histogram

Reply:
I'm probably getting this wrong, but shouldn't be all histograms valid in order to be considered by the linear combination? Set status to 1 means that the file (or histogram) is not found, which I'm checking in line 921. I will add a check here. If branching ratio is 0.0, I will ignore the missing histogram and don't force gLike to break. If the histogram is not found and branching ratio != 0.0, I will set status to 1, which force gLike to break, which is necessary in my point of view. Please instruct if I'm wrong!

Reply:
Ups, sorry, I always tend to think 0 is unscess, but it's 1, you are right...